src/ReaderFactory.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Cycle\Annotated;
+
+use Doctrine\Common\Annotations\AnnotationReader as DoctrineAnnotationReader;
+use Spiral\Attributes\AnnotationReader;
+use Spiral\Attributes\AttributeReader;
+use Spiral\Attributes\Composite\SelectiveReader;
+use Spiral\Attributes\ReaderInterface;
+
+final class ReaderFactory
+{
+    /**
+     * @param ReaderInterface|DoctrineAnnotationReader|null $reader
+     * @return ReaderInterface
+     *
+     * @psalm-type ReaderType = ReaderInterface | DoctrineAnnotationReader | null
+     * @psalm-param ReaderType $reader
+     */
+    public static function create($reader = null): ReaderInterface
```

Comment:
The semantic purpose of the word `create` is to create a new object. Therefore, in the original version, I preferred him a more close to reality `resolve`.